lib/collection/src/collection/distance_matrix.rs

```diff
@@ -245,17 +245,15 @@ impl Collection {
         // postprocess the results to account for overlapping samples
         for (scores, sample_id) in nearest.iter_mut().zip(sampled_point_ids.iter()) {
             // need to remove the sample_id from the results
-            let mut filtered: Vec<_> = scores
-                .iter()
-                .filter(|p| p.id != *sample_id)
-                .cloned()
-                .collect();
-            // if not found pop lowest score
-            if filtered.len() == limit_per_sample + 1 {
-                // if we have enough results, remove the last one
-                filtered.pop();
+            if let Some(sample_pos) = scores.iter().position(|p| p.id == *sample_id) {
+                scores.remove(sample_pos);
+            } else {
+                // if not found pop lowest score
+                if scores.len() == limit_per_sample + 1 {
+                    // if we have enough results, remove the last one
+                    scores.pop();
+                }
             }
-            *scores = filtered;
```

Comment:
The new behavior is different, is that correct?

Reply:
How is it different?

Reply:
The new implementation pops an item at a specific position, the previous implementation did not.

Reply:
I see now, the previous implementation did this as well by reconstructing a filtered list.

Your new approach is better, thanks!